lms/djangoapps/mobile_api/video_outlines/tests.py

```diff
@@ -206,6 +216,42 @@ def test_course_list_with_nameless_unit(self):
         self.assertEqual(len(course_outline), 1)
         self.assertEqual(course_outline[0]['path'][2]['name'], self.nameless_unit.location.block_id)
 
+    def test_course_list_with_video_in_sub_section(self):
+        """
+        Tests a non standard xml format where a video is underneath a sequential
+
+        We are expecting to return the same unit and section url since there is
+        no unit vertical.
+        """
+        ItemFactory.create(
+            parent_location=self.sub_section.location,
+            category="video",
+            edx_video_id=self.edx_video_id,
+            display_name=u"test draft video omega 2 \u03a9"
+        )
+        course_outline = self._get_video_summary_list()
+        self.assertEqual(len(course_outline), 1)
+        self.assertEqual(len(course_outline[0]['path']), 2)
+        self.assertEqual(course_outline[0]["section_url"], course_outline[0]["unit_url"])
+
+    def test_course_list_with_split_test(self):
+        ItemFactory.create(
+            parent_location=self.split_test.location,
+            category="video",
+            display_name=u"split test vertical a",
+        )
+        ItemFactory.create(
+            parent_location=self.split_test.location,
+            category="video",
+            display_name=u"split test vertical b",
+        )
+        course_outline = self._get_video_summary_list()
+        self.assertEqual(len(course_outline), 2)
+        self.assertEqual(len(course_outline[0]["path"]), 4)
+        self.assertEqual(len(course_outline[1]["path"]), 4)
+        self.assertEqual(course_outline[0]["summary"]["name"], u"split test vertical a")
+        self.assertEqual(course_outline[1]["summary"]["name"], u"split test vertical b")
+
```

Comment:
No tests for `chapter` being None or `section` being None - since the new code should be handling this? 
Otherwise, the `else` clauses of L68-69 are not tested: https://github.com/edx/edx-platform/pull/6442/files#diff-de0e3088a167332527d38a99af73ea19R68